packages/functions/src/getPlayerMatches.ts

```diff
@@ -62,14 +62,16 @@ const getPlayerMatches = functions
   .runWith(runtimeOpts)
   .pubsub.topic(PUBSUB_TOPIC_DOWNLOAD_MATCHES)
   .onPublish(async (message) => {
-    const profileNames = message.json.profileNames;
+    const profileNames: Array<string> = message.json.profileNames;
     functions.logger.log(`Received these profile names ${profileNames}`);
 
     const matches: Record<string, any> = {};
     let duplicatesCounter = 0;
 
-    for (const profileName of profileNames) {
-      const playerMatches = await getAndPrepareMatchesForPlayer(profileName);
+    const chunkedData = chunk(profileNames, 5);
```

Comment:
Unfornatelly sending 5 profiles instead of 1 takes exactly the same time as if we would send 5x requests. The bottle neck is in the DB probably.